index.js

```diff
@@ -1,20 +1,55 @@
 var execFile = require('child_process').execFile,
-    path = require('path'),
-    win32 = require('os').platform() === 'win32';
+    fs = require('fs'),
+    os = require('os'),
+    path = require('path');
+
+/**
+ * Locates the JSDoc executable command.  Since a module is not provided,
+ * `require.resolve('jsdoc')` does not work and must be done manually.
+ *
+ * @param {String}  dir The starting directory to search.
+ * @return {String} The executable path, or null if not found.
+ */
+function locateJSDocCommand(dir) {
+  var executable = os.platform() === 'win32' ? 'jsdoc.cmd' : 'jsdoc',
+      cmd;
+  dir = path.resolve(dir);
+  while (dir) {
+    try {
+      cmd = path.join(dir, 'node_modules', '.bin', executable);
+      // End the search if the command is found.
+      // If not found, an exception is thrown.
+      fs.statSync(cmd);
+      break;
+    } catch (ex) {
+      cmd = null;
+      // Otherwise, iterate to the parent directory, if possible.
+      if (path.dirname(dir) == dir) {
+        dir = null;
+      } else {
+        dir = path.resolve(path.dirname(dir));
```

Comment:
Non-blocker: Could move `path.resolve(dir)` from outside the loop to the first part of the loop to avoid re-resolving here. Could then just set `dir` to `path.dirname(dir)`